Python_DataEngineering/Database.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function
+
+import mysql.connector
+from mysql.connector import errorcode
+
+DB_NAME = 'BikePoints'
+
+TABLES = {}
+TABLES['bikes_availablility'] = (
+    "CREATE TABLE `bikes_availability` ("
+    "  `bikeid` int(11) NOT NULL,"
+    "  `nam` varchar(50) NOT NULL,"
+    "  `lat` varchar(50) NOT NULL,"
+    "  `longitude` varchar(50) NOT NULL,"
+    "  `installed` enum('true','false') NOT NULL,"
+    "  `locked` enum('true','false') NOT NULL,"
+    "   `nbBikes` int(3),"
+    "   `nbEmptyDocks`  int(3),"
+    "  PRIMARY KEY (`bikeid`)"
+    ")")
+
+TABLES['timestamp_bikes_data'] = (
+    "CREATE TABLE `timestamp_bikes_data` ("
+    "  `bikeid` int(11) NOT NULL,"
+    "  `nam` varchar(50) NOT NULL,"
+    "  `lat` varchar(50) NOT NULL,"
+    "  `longitude` varchar(50) NOT NULL,"
+    "  `installed` enum('true','false') NOT NULL,"
+    "  `locked` enum('true','false') NOT NULL,"
+    "   `TimeSnap` datetime,"
+    "   `nbBikes` int(3),"
+    "   `nbEmptyDocks`  int(3),"
+    "  PRIMARY KEY (`TimeSnap`)"
```

Comment:
This might create problems - what if 2 bike stands are being updated at the same time? If your primary key is only timestamp, you can only store 1 of them

Reply:
Yes, that is true. I removed the primary key as TimeSnap and added a unique constraint of BikeId and TimeSnap. Different combination of bikeId and TimeSnap will create new rows. It generally means new rows are not added until and unless it is either a new bikeId or an old bikeId with latest data.